### PR TITLE
[Bugfix] For Value Relation, Empty value is added for required fields

### PR DESCRIPTION
**Pull request by rldhont**
Lizmap added an empty value to the value relation list if the field is required. This has been removed form QgisFormValueRelationDynamicDatasource.

* Funded by 3liz

### CODE REVIEW

**Comment by rldhont** [comment]
Cypress tests are green

**Comment by rldhont** [comment]
For 3.4.* the fix has to be done here https://github.com/3liz/lizmap-web-client/blob/release_3_4/lizmap/modules/lizmap/classes/qgisForm.class.php#L1306

**Review by mdouchin — APPROVED** [review]

**Comment by rldhont** [comment]
With last changes, cypress failed:
```
Advanced form
    ✓ should toggle tab visibility when toggling checkbox (6158ms)
    1) should have expression constraint on field
    2) should change selected quartier and sousquartier based on drawn point


  1 passing (24s)
  2 failing

  1) Advanced form
       should have expression constraint on field:
     AssertionError: Timed out retrying after 4000ms: expected '<label#jforms_view_edition_website_label.control-label.jforms-label>' to have class 'jforms-error'
      at Context.eval (http://localhost:8130/__cypress/tests?p=cypress/integration/form_advanced-ghaction.js:127:50)

  2) Advanced form
       should change selected quartier and sousquartier based on drawn point:

      AssertionError: Timed out retrying after 4000ms: Too many elements found. Found '3', expected '1'.
      + expected - actual

      -3
      +1
      
      at Context.eval (http://localhost:8130/__cypress/tests?p=cypress/integration/form_advanced-ghaction.js:140:52)
```

**Comment by nboisteault** [comment]
@rldhont you can then test manually to understand the issue. 

Here is behavior with you PR :
bug_PR_2783

Correct behavior with master :

no_bug_PR_2783

**Comment by rldhont** [comment]
Thanks @nboisteault can you check GetDataList requests ?

File: tests/end2end/cypress/integration/form_advanced-ghaction.js [hunk 1]
```diff
@@ -67,9 +68,11 @@ describe('Advanced form', function () {
 
         // Assert quartier value is good for another drawn point
         cy.get('#map').click(600, 350)
-        cy.get('#jforms_view_edition_quartier option').should('have.text', 'MONTPELLIER CENTRE')
+        cy.get('#jforms_view_edition_quartier option').should('have.length', 2)
+        cy.get('#jforms_view_edition_quartier option').last().should('have.text', 'MONTPELLIER CENTRE')
+        cy.get('#jforms_view_edition_quartier').select('MC')
```
Comment (nboisteault) [inline tests/end2end/cypress/integration/form_advanced-ghaction.js:73]: @rldhont In QGIS user does not have to select. Could we have the same behavior in Lizmap?
Peek 17-02-2022 20-38

**Comment by rldhont** [comment]
With last changes all cypress tests are green ! I will rebase the branch to clean commits

**Review by laurentj — APPROVED** [review]
I'm not agree with the adding of an empty value when the control is required. Except that UX issue, LGTM

**Pull request merged**